### PR TITLE
Catalog Ingestion Product Event Manager Implementation

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Product Event Manager class implementation for manipulating product event data in magento.
Main Request Builder class implementation to build "v1/catalog/m2/product" request.

https://app.asana.com/0/1201931884901947/1201906341917644/f

#changelog Catalog Ingestion Product Event Manager Implementation

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by garmashdev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1559** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1559" target="_blank">Graphite</a> 
    * **PR #1560** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1560" target="_blank">Graphite</a>  👈
      * **PR #1561** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1561" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1560?utm_source=stack-comment).

File: Model/CatalogIngestion/Request/Product/DataProcessor.php [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+                'Values' => [],
+                'Visibility' => 'visible',
+                'SortOrder' => $option->getSortOrder(),
+            ];
+
+            if ($values = $option->getValues()) {
+                foreach ($values as $valueId => $value) {
+                    $optionData['values'][] = [
+                        'value' => $valueId,
+                        'display_value' => $value->getTitle(),
```
Comment (l2yao) [inline Model/CatalogIngestion/Request/Product/DataProcessor.php:303]: still snake case, lets change them

Comment (garmashdev) [reply Model/CatalogIngestion/Request/Product/DataProcessor.php]: Updated

File: Api/ProductEventManagerInterface.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+interface ProductEventManagerInterface
+{
+    /**
+     * Publish new product event
+     *
+     * @param int $productId
+     * @param string $type
+     * @return ProductEventInterface
+     */
+    public function publishProductEvent(int $productId, string $type): ProductEventInterface;
```
[the remark this answers is not in the document]
Comment (vitaliyreznikov) [reply Api/ProductEventManagerInterface.php:37]: I think it's ok, at least for now.

File: Api/ProductEventManagerInterface.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Request catalog product event to bolt
+     *
+     * @param ProductEventInterface $productEvent
+     * @return bool
+     * @throws LocalizedException
+     * @throws \Zend_Http_Client_Exception
+     */
+    public function requestProductEvent(ProductEventInterface $productEvent): bool;
```
Comment (vitaliyreznikov) [inline Api/ProductEventManagerInterface.php:66]: Let's rename it to "sendProductEvent". You can do it in separate PR, just to reduces changes in the existing PRs

File: Model/CatalogIngestion/ProductEventProcessor.php [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+        $this->productFactory = $productFactory;
+        $this->productWebsiteLink = $productWebsiteLink;
+        $this->eavConfig = $eavConfig;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Update product events based on source items
+     *
+     * @param array $sourceItems
```
[the remark this answers is not in the document]
Comment (vitaliyreznikov) [reply Model/CatalogIngestion/ProductEventProcessor.php:90]: And improve function descriptions here and below as well.

[the remark this answers is not in the document]
Comment (garmashdev) [reply Model/CatalogIngestion/ProductEventProcessor.php]: Updated.

**Review by vitaliyreznikov — APPROVED** [review]
